### PR TITLE
Modify for Amazon Linux 2023

### PR DESCRIPTION
**Pull request by yuichisatake**
python3 is installed by default on amazon linux2023.
So, when I changed ansible's interpreter_python to "/usr/bin/python3" and executed the role, the following error occurred.

```
TASK [mackerelio.mackerel-agent : install mackerel-agent] ****************************************************************************************************
task path: /home/sysadmin/.ansible/roles/mackerelio.mackerel-agent/tasks/amazonlinux.yml:31
fatal: [10.16.4.68]: FAILED! => {"changed": false, "msg": "The Python 2 yum module is needed for this module. If you require Python 3 support use the `dnf` Ansible module instead."}`
```

I created a pull request that works with the pyton3 interpreter, so please consider merging it.

### CODE REVIEW

**Comment by lufia** [comment]
I see, `yum` module is strongly depends on Python 2.

We are trying to merge this PR onto main branch, however, I have some questions.

1. Could you add `dnf` support to files under **tasks** directory, such as **amazonlinux.yml** or **main.yml**?
2. If can't, are there technical issues?

**Comment by yuichisatake** [comment]
@lufia 
sorry. The location of the task file was incorrect.
We have corrected it, so please check it.

File: files/amazonlinux-mackerel-v2023.repo
[the remark this answers is not in the document]
Comment (yuichisatake) [reply files/amazonlinux-mackerel-v2023.repo:1]: Sorry,it is same. This has been corrected.

**Comment by yuichisatake** [comment]
@lufia 
I have fixed the pull request, so please check it.

**Comment by lufia** [comment]
@yuichisatake Thank you!

I'm going to merge this patch.

**Review by lufia — APPROVED** [review]

**Pull request merged**